h/util/metrics.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import unicode_literals
+import newrelic.agent
+
+
+def send_metric(name, metric):
+    """ Send a metric to New Relic.
+
+    :arg name: The key name. Only the first 255 characters are retained
+    :type name: str
+    :arg metric: The string value to add to the current transaction
+    Only the first 255 characters are retained.
+    :type metric: str, int, float, bool
+    """
+    newrelic.agent.add_custom_parameter(name, metric)
+
+
+def record_search_query_param_usage(params, separate_replies):
+    """ Send metric for each parameter if it matches list of desired keys.
+
+    :arg params: the search parameters that will be popped by each of the filters
```

Comment:
```suggestion
    :arg params: the request params to record
```

We shouldn't say "popped" here because popping any item from a dict (with the `.pop()` method) removes it from the dict, which isn't what's happening here. Let's not say "filter" either as that's an internal implementation detail.